stripe/resource.py

```diff
@@ -393,7 +393,9 @@ def list(cls, api_key=None, idempotency_key=None,
         requestor = api_requestor.APIRequestor(api_key, account=stripe_account)
         url = cls.class_url()
         response, api_key = requestor.request('get', url, params)
-        return convert_to_stripe_object(response, api_key, stripe_account)
+        stripe_object = convert_to_stripe_object(response, api_key, stripe_account)
+        stripe_object._retrieve_params = params
```

Comment:
While I'm never a fan of assigning to "private" instance variables, I believe this is the best way to fix the current bug.